src/xss-filters.js

```diff
@@ -24,102 +24,39 @@ exports._getPrivFilters = function () {
-                '(?:a|A|&#[xX]0*(?:6|4)1;?|&#0*(?:97|65);?)',
-
-            ].join(URI_BLACKLIST_INTERIM_WHITESPACE),
-            // vb
-            [
-                '(?:v|V|&#[xX]0*(?:7|5)6;?|&#0*(?:118|86);?)',
-                '(?:b|B|&#[xX]0*(?:6|4)2;?|&#0*(?:98|66);?)'
-
-            ].join(URI_BLACKLIST_INTERIM_WHITESPACE)
-
-        ].join('|'),
-        ')',
-
-        URI_BLACKLIST_INTERIM_WHITESPACE,
-
-        // script:
-        [
-            '(?:s|S|&#[xX]0*(?:7|5)3;?|&#0*(?:115|83);?)',
-            '(?:c|C|&#[xX]0*(?:6|4)3;?|&#0*(?:99|67);?)',
-            '(?:r|R|&#[xX]0*(?:7|5)2;?|&#0*(?:114|82);?)',
-            '(?:i|I|&#[xX]0*(?:6|4)9;?|&#0*(?:105|73);?)',
-            '(?:p|P|&#[xX]0*(?:7|5)0;?|&#0*(?:112|80);?)',
-            '(?:t|T|&#[xX]0*(?:7|5)4;?|&#0*(?:116|84);?)',
-            '(?::|&#[xX]0*3[aA];?|&#0*58;?|&colon;)'
-
-        ].join(URI_BLACKLIST_INTERIM_WHITESPACE)
-    ].join('');
-    */
-
-    var URI_SLOWLANE = ['&', 'j', 'J', 'v', 'V'],
-        // delay building URI_BLACKLIST as an RegExp() object until the first hit
-        // the following str is generated by the above commented logic
-        URI_BLACKLIST_REGEXP = /^(?:&#[xX]0*(?:1?[1-9a-fA-F]|10|20);?|&#0*(?:[1-9]|[1-2][0-9]|30|31|32);?|&Tab;|&NewLine;)*(?:(?:j|J|&#[xX]0*(?:6|4)[aA];?|&#0*(?:106|74);?)(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?:a|A|&#[xX]0*(?:6|4)1;?|&#0*(?:97|65);?)(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?:v|V|&#[xX]0*(?:7|5)6;?|&#0*(?:118|86);?)(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?:a|A|&#[xX]0*(?:6|4)1;?|&#0*(?:97|65);?)|(?:v|V|&#[xX]0*(?:7|5)6;?|&#0*(?:118|86);?)(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?:b|B|&#[xX]0*(?:6|4)2;?|&#0*(?:98|66);?))(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?:s|S|&#[xX]0*(?:7|5)3;?|&#0*(?:115|83);?)(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?:c|C|&#[xX]0*(?:6|4)3;?|&#0*(?:99|67);?)(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?:r|R|&#[xX]0*(?:7|5)2;?|&#0*(?:114|82);?)(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?:i|I|&#[xX]0*(?:6|4)9;?|&#0*(?:105|73);?)(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?:p|P|&#[xX]0*(?:7|5)0;?|&#0*(?:112|80);?)(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?:t|T|&#[xX]0*(?:7|5)4;?|&#0*(?:116|84);?)(?:&#[xX]0*[9aAdD];?|&#0*(?:9|10|13);?|&Tab;|&NewLine;)*(?::|&#[xX]0*3[aA];?|&#0*58;?|&colon;)/;
-
-    // Given a full URI, need to support "[" ( IPv6address ) "]" in URI as per RFC3986
-    // Reference: https://tools.ietf.org/html/rfc3986
-    var URL_IPV6 = /\/\/%5[Bb]([A-Fa-f0-9:]+)%5[Dd]/;
+    var URI_BLACKLIST_PROTOCOLS = ['javascript', 'data', 'vbscript'],
+        URI_PROTOCOL_COLON = /(?::|&#[xX]0*3[aA];?|&#0*58;?|&colon;)/,
+        URI_PROTOCOL_HTML_ENTITIES = /&(?:#([xX][0-9A-Fa-f]+|\d+);?|Tab;|NewLine;)/g,
+        URI_PROTOCOL_WHITESPACES = /(?:^[\u0001-\u0020]+|[\t\n\r]+)/g,
```

Comment:
is it include all whitespaces?
https://developer.mozilla.org/en/docs/Web/JavaScript/Guide/Regular_Expressions#special-white-space